src/frequenz/client/reporting/_client.py

```diff
@@ -200,10 +200,12 @@ def sample(self) -> MetricSample:
 class ReportingApiClient(BaseApiClient[ReportingStub]):
     """A client for the Reporting service."""
 
+    # pylint: disable=too-many-arguments, too-many-positional-arguments
```

Comment:
If you use `disable=` in its own line, that will disable the check for the rest of the file, not only for the next line, if you want to disable it only for the next line you need to use `disable-next=`.

```suggestion
    # pylint: disable-next=too-many-arguments, too-many-positional-arguments
```

Also I think in this case `pylint` has a point, and you should use positional-only arguments, specially for strings/bool that are really hard to read without a keyword on the call site.

Reply:
Again, me being to meek about breaking the interface, but I am already doing it anyways.

Reply:
We are at 0.x, so...

![BREAK ALL THE THINGS!](https://github.com/user-attachments/assets/5fbe86a9-ed89-4358-bfbf-aebe9411311e)